### PR TITLE
Add setup.py file

### LINKED ISSUE #2: no setup.py
Repository version has no setup.py

### PR DESCRIPTION
**Pull request by nswarup14**
Fixes #2 
Thanks

### CODE REVIEW

File: setup.py
Comment (rhl-bthr) [inline setup.py]: Please add missing newline at the end of the file

**Review by rhl-bthr — CHANGES_REQUESTED** [review]
See review comments

**Comment by nswarup14** [comment]
@pro-panda 
Added the newline
Please review, thanks

**Comment by quozl** [comment]
For interest;

```
% python setup.py dist_xo
WARNING:root:Activity bundle /tmp/riverhex-activity does not specify a license
WARNING:root:Missing po/ dir, cannot build_locale
```

I'm not sure if the license covers the old 32-bit binary `bin/riverhexactivity.exe`.

File: setup.py
Comment (nswarup14) [reply setup.py]: Done

**Comment by nswarup14** [comment]
@quozl 
When I run `sudo python setup.py dist_xo`, I do not get any warnings.
I did not understand what you pointed about the old 32-bit binary executable and its license.

Please advise, Thanks

**Comment by quozl** [comment]
> sudo ... do not get any warnings

Do not use `sudo` to run `dist_xo`; there's no need for it.  You will not get warnings if you are using an old version of the Sugar Toolkit.  What version are you using?

> I did not understand what you pointed about the old 32-bit binary executable and its license.

Happy to explain.

You added a  file.  This enabled tests that were not enabled before.  As a result of the tests, I saw there is a 32-bit binary file.  The path is as described.  To comply with license (see `LICENSE`, and `activity.py`), we must be able to produce the source.  I can't find the source in the repository.  This makes the bundle produced by `dist_xo` impossible to distribute by Sugar Labs, due to our nature and governance.  That it is 32-bit makes the activity architecture specific.  It may also make the activity not work on 64-bit architectures and different CPUs such as ARM.  I've not tested the activity; have you?  These are ancillary problems; but far greater priority than adding a `setup.py` file.  I'd prefer to see a complete solution before merging.

**Comment by nswarup14** [comment]
> Do not use sudo to run dist_xo; there's no need for it.

@quozl , I tried running without sudo, but it showed me a `permission denied` message. Does that mean something is incorrect on my system.
I am working on a Lenovo, 64-bit thinkpad with Ubuntu 18.04 and Sugar v0.112 (I think the latest version it self).

> It may also make the activity not work on 64-bit architectures and different CPUs such as ARM

The activity is architecture specific, because in the `activity.py`, its actually mentioned that the activity wont run on ARM architecture. And no, I haven't been able to test on other architectures because I lack them.

> I can't find the source in the repository

You're right, there is no source code and hence by the `LICENSE` cannot be distributed.

Is there anything I could do to help with this?
Thank you!

**Comment by quozl** [comment]
> is there anything I could do to help with this?

Possibly.  See if you can move a step closer after this research;

* on our mailing lists there is almost no mention of RiverHex, one exception is a test report by Tony Anderson on 29th December 2015,
* according to http://activities.sugarlabs.org/en-US/sugar/addon/4325 the maintainer was @llaske in 2010, and the activity is written in Mono, see http://wiki.sugarlabs.org/go/Mono for technical detail,
* there's no repository that I've found on GitHub or git.sugarlabs.org that contains the source code,
* there's nothing on a Google Search that seems to fit,
* the bundle file riverhex-2.xo available from activities.sugarlabs.org has shareable library files in `bin/` that are critical, but they have been lost in the transition from bundle to GitHub, by @cristian99garcia,

Perhaps next step is just to ask questions of appropriate people; @llaske, have you the source code?

> but it showed me a permission denied message

Mine does not show that message.  Is there any further detail, like a file name?  You might try `strace -e open` prefix to the command to get a log of all files opened and result of open attempt.   Or check all the files in the repository have you as owner rather than `root`.

> 0.112 (I think the latest version it self)

Thanks.  Yes, 0.112 is the latest version available from Ubuntu, but we have commits to Sugar and the Toolkit since.  The messages I see are probably related to changes since 0.112.

**Comment by llaske** [comment]
I'm the author of the activity but I must confess I've stopped to maintain it for a long time.
I've uploaded all source code I've found on my laptop here: https://github.com/llaske/RiverHexActivity
Hope it will help.

**Comment by quozl** [comment]
Thanks @llaske!

**Comment by quozl** [comment]
Closing pull request.  Fix https://github.com/sugarlabs/riverhex-activity/issues/5 first, thanks.

**Pull request closed without merging**
